packages/front-end/components/Features/FeatureDiagnostics.tsx

```diff
@@ -0,0 +1,314 @@
+        // add an id to each row based on the timestamp
+        const rowsWithId = results.rows.map((row) => ({
+          ...row,
+          id: row.timestamp ?? "",
+        }));
+        setResults(rowsWithId);
+      } else {
+        setResults([]);
+      }
+    } catch (e) {
+      setError(e.message);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  // Empty State: Prompt user to set up a data source to view diagnostics for this feature
+  if (validDatasources.length === 0) {
+    return (
+      <Box className="contents container-fluid pagecontents">
+        <EmptyState
+          title="Feature Evaluation Diagnostics"
+          description="Set up a data source to view diagnostics for this feature."
+          leftButton={
+            <LinkButton href="/datasources">Set up a Data Source</LinkButton>
+          }
+          rightButton={null}
+        />
+      </Box>
+    );
+  }
+
+  return (
+    <Box className="contents container-fluid pagecontents">
+      <h2>Feature Evaluation Diagnostics</h2>
+      <p>
+        View recent feature evaluations along with any custom metadata
+        you&apos;ve added.
+      </p>
+
+      {datasource && !datasourceHasFeatureUsageQuery && (
+        <Callout status="info" mb="4">
+          Feature Evaluation Diagnostics require setting up a feature usage
+          query in your data source.
+          <Link href={`/datasources/${datasource.id}`} ml="2">
+            Setup a Feature Usage Query
+          </Link>
+        </Callout>
+      )}
+      <Box width="400px">
+        <SelectField
+          label="Select a Data Source"
+          labelClassName="font-weight-bold"
+          value={form.watch("datasourceId") ?? ""}
+          onChange={(newDatasource) => {
+            form.setValue("datasourceId", newDatasource);
+          }}
+          options={validDatasources.map((d) => {
+            const isDefaultDataSource = d.id === settings.defaultDataSource;
+            return {
+              value: d.id,
+              label: `${d.name}${
+                d.description ? ` — ${d.description}` : ""
+              }${isDefaultDataSource ? " (default)" : ""}`,
+            };
+          })}
+          className="portal-overflow-ellipsis"
+        />
+      </Box>
+
+      <Frame mt="4">
+        <Flex direction="row" justify="end" mb="4">
+          <Tooltip
+            content="Setup a feature usage query in your data source to view feature evaluations."
+            enabled={!datasourceHasFeatureUsageQuery && !loading}
+          >
+            <Button
+              onClick={onRunFeatureUsageQuery}
+              disabled={loading || !datasourceHasFeatureUsageQuery}
+            >
+              {loading
+                ? "Running..."
+                : !results
+                  ? "View recent feature evaluations"
+                  : "Refresh feature evaluations"}
+            </Button>
+          </Tooltip>
+        </Flex>
+        {error && (
```

Comment:
issue: I was getting an error but it was not surfaced here. Need to fix it.

Here's the full response from `POST /query/feature-eval-diagnostic`:
```
{
  "status": 200,
  "error": "Every derived table must have its own alias",
  "sql": "-- Feature Evaluation Diagnostics Query\nSELECT\n  *\nFROM\n  (\n    SELECT\n      *\n    from\n      `dummy-data`.`feature_events`\n    WHERE\n      feature_key = 'feature-diagnostics-test'\n  )\nORDER BY\n  timestamp DESC\nLIMIT\n  1000"
}
```